src/afrolabs/config.clj

```diff
@@ -140,6 +140,17 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(defmethod aero/reader 'config/case
+  [_opts _ value]
+  (when-not (odd? (count value))
+    (throw (ex-info "Provide the case value and the cases as pairs." {})))
+  (let [[the-case & cases] value
+        all-cases (into {} (map vec (partition 2 cases)))]
+    (get all-cases the-case) ;; might return nil...
+    ))
```

Comment:
This beauty allows for aero config like this:

```clojure
{:bootstrap-server #or [#config/case [#or [#option KAFKA_AUTH_PROVIDER "Confluent"]
                                       "Confluent" #option confluent_kafka_bootstrap_server
                                       "MskIam"    #option aws_msk_iam_bootstrap_server
                                       "MskScram"   #option aws_msk_scram_bootstrap_server]
                         #option KAFKA_BOOTSTRAP_SERVER]
  :auth/strategy    #config/case [#or [#option KAFKA_AUTH_PROVIDER
                                       "Confluent"]
                                  "Confluent" [:strategy/ConfluentCloud
                                               :api-key    #option CONFLUENT_API_KEY
                                               :api-secret #option CONFLUENT_API_SECRET]
                                  "MskIam"    [:strategy/AwsMskIam]
                                  "MskScram"  [:strategy/AwsMskScram
                                               :aws.msk/username #option aws_msk_scram_username
                                               :aws.msk/password #option aws_msk_scram_password]]
  }
```

Reply:
This will read an environment variable `KAFKA_AUTH_PROVIDER` and choose which auth strategy to implement, or which `:bootstrap-server` to use. This is then followed later with `#ref` refenerences, like this:

```clojure
:afrolabs.gometro-bridge.core/ktable-asserter:config
 {:bootstrap-server                 #ref [:afrolabs.gometro-bridge.core/common-kafka-config :bootstrap-server]
  :strategies                       [#ref [:afrolabs.gometro-bridge.core/common-kafka-config :auth/strategy]]
 ...
   }
```